R/matching.R

```diff
@@ -147,6 +151,114 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
+    hist_plot <- hist_plot + ggplot2::labs(caption = stringr::str_wrap(print_text, width = caption_width)) +
+      ggplot2::theme(
+        plot.caption.position = "plot",
+        plot.caption = element_text(hjust = 0)
+      )
+  }
+  return(hist_plot)
+}
+
+
+#' Plot method for Estimate Weights objects
+#'
+#' The plot function displays individuals weights with key summary in top right legend that includes
+#' median weight, effective sample size (ESS), and reduction percentage (what percent ESS is reduced from the
+#' original sample size). There are two options of plotting: base R plot and ggplot. The default
+#' for base R plot is to plot unscaled and scaled separately. The default
+#' for ggplot is to plot unscaled and scaled weights on a same plot.
+#'
+#' @param x object from [estimate_weights]
+#' @param ggplot indicator to print base weights plot or ggplot weights plot
+#' @param bin_col a string, color for the bins of histogram
+#' @param vline_col a string, color for the vertical line in the histogram
+#' @param main_title title of the plot. For ggplot, name of scaled weights plot and unscaled weights plot, respectively.
+#' @param scaled_weights (base plot only) an indicator for using scaled weights instead of regular weights
+#' @param bins (ggplot only) number of bin parameter to use
+#' @param print_caption (ggplot only) print a footnote message related to ESS from the NICE survey 2021
+#' @param caption_width (ggplot only) width that is passed onto str_wrap function
+#'
+#' @examples
+#' @describeIn estimate_weights Plot method for estimate_weights objects
+#' @export
+
+plot.maicplus_estimate_weights <- function(x, ggplot = FALSE,
+                                           bin_col = NULL, vline_col = NULL,
+                                           main_title = NULL, scaled_weights = TRUE,
+                                           bins = 50, print_caption = FALSE,
+                                           caption_width = 80) {
+  if (!ggplot) {
+    if (is.null(main_title)) main_title <- "Scaled Individual Weights"
+    if (is.null(bin_col)) bin_col <- "#6ECEB2"
+    if (is.null(vline_col)) vline_col <- "#688CE8"
+  } else {
+    if (is.null(main_title)) main_title <- c("Scaled Individual Weights", "Unscaled Individual Weights")
+    if (is.null(bin_col)) bin_col <- "black"
+    if (is.null(vline_col)) vline_col <- "red"
+  }
+
+  if (!ggplot) {
+    plot_weights_base(x, bin_col, vline_col, main_title, scaled_weights)
+  } else {
+    plot_weights_ggplot(x, bin_col, vline_col, main_title, bins, print_caption, caption_width)
+  }
+}
```

Comment:
You could combine this into one `if (isTRUE(ggplot)){ ... } else { ... }`

Reply:
combined